yazi: improve fish integration

iSma opened the pull request:
### Description

Improve yazi fish integration:

Calls yazi as `command yazi`, allowing to use "yazi" as `shellWrapperName` without infinite recursion. This could be backward-incompatible for users who have an alias named "yazi" and expect this shell wrapper to call that alias instead of the yazi executable directly.

Also defines the wrapper with `programs.fish.functions` instead of `interactiveShellInit`, which seems to be the preferred way.

### Checklist

- [x] Change is backwards compatible. (with one exception, see above)

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@XYenon @eljamm

eljamm commented:
> This could be backward-incompatible for users who have an alias named "yazi" and expect this shell wrapper to call that alias instead of the yazi executable directly.

Could you explain this, perhaps? To me, it seems that we're doing the same thing: creating a function for the integration, but now we're storing it in `functions` instead of including it directly in the `config`.

eljamm commented on tests/modules/programs/yazi/fish-integration-enabled.nix:
```diff
@@ -1,27 +1,18 @@
 
   nmt.script = ''
-    assertFileContains home-files/.config/fish/config.fish '${shellIntegration}'
+    assertFileContent home-files/.config/fish/functions/yy.fish \
```
I'd personally prefer if we don't hardcode the function name, here, since the test can break if it's not changed when the `shellWrapperName` does.

```suggestion
    assertFileContent home-files/.config/fish/functions/${config.programs.yazi.shellWrapperName}.fish \
```

iSma replied:
Good point, I've amended the commit with this change.

iSma commented:
> > This could be backward-incompatible for users who have an alias named "yazi" and expect this shell wrapper to call that alias instead of the yazi executable directly.
> 
> Could you explain this, perhaps? To me, it seems that we're doing the same thing: creating a function for the integration, but now we're storing it in `functions` instead of including it directly in the `config`.

Yes, that part doesn't change; the end result is that we create a function named `$shellWrapperName`.

The difference I was pointing to is the way yazi is called from within this function: I've changed it to `command yazi` which calls executable directly and ignores potential shell aliases named "yazi". Previously, a user could potentially define `alias yazi=yazi --some-options` and the wrapper would call this alias.

I don't think the old behavior is particularly useful, since users can (and probably should) set their options directly in the settings. But I still wanted to point it out for the sake of completeness.

eljamm commented:
> The difference I was pointing to is the way yazi is called from within this function: I've changed it to command yazi which calls executable directly and ignores potential shell aliases named "yazi".

Ah, I see. I missed that, somehow.

Since this might be a breaking change, do you think it's better if we write something in the news file, like this perhaps:

https://github.com/nix-community/home-manager/blob/fc52a210b60f2f52c74eac41a8647c1573d2071d/modules/misc/news.nix#L1679-L1690

But just for fish integration users.

iSma commented:
> > The difference I was pointing to is the way yazi is called from within this function: I've changed it to command yazi which calls executable directly and ignores potential shell aliases named "yazi".
> 
> Ah, I see. I missed that, somehow.
> 
> Since this might be a breaking change, do you think it's better if we write something in [the news file](https://nix-community.github.io/home-manager/index.xhtml#sec-news), like this perhaps:
> 
> https://github.com/nix-community/home-manager/blob/fc52a210b60f2f52c74eac41a8647c1573d2071d/modules/misc/news.nix#L1679-L1690
> 
> But just for fish integration users.

Yes to be on the safe side I added a news entry.

rycee commented:
Thanks! Merged to master now 🙂
